scripts/zones/Dynamis-Xarcabard/mobs/Vanguard_Eye.lua

```diff
@@ -35,27 +35,48 @@ end;
 -----------------------------------
 
 function onMobDeath(mob,killer)
-	
-	local mobID = mob:getID();
-	
-	-- 035 039: spawn 043 when defeated
-	if (mobID == 17326536) then
-		SpawnMob(17326553);
-	-- 058: spawn 60 when defeated
-	elseif (mobID == 17326661 and GetMobAction(17326668) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326668 and GetMobAction(17326661) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326673 and GetMobAction(17326661) == 0 and GetMobAction(17326668) == 0) then
-		SpawnMob(17326706);
-	-- 114: spawn 112 when defeated
-	elseif (mobID == 17326790) then
-		SpawnMob(17326086);
-	-- 144-149: spawn 150 when defeated
-	elseif (mobID >= 17330913 and mobID <= 17330918) then
-		SetServerVariable("[DynaXarcabard]TE150_Trigger",GetServerVariable("[DynaXarcabard]TE150_Trigger") + (mobID - 17330912) ^ 2);
-	end
-	
-	if (GetServerVariable("[DynaXarcabard]TE150_Trigger") == 63) then
-		SpawnMob(17330919); -- 150
-	end
-	
-end;
+    
+    local mobID = mob:getID();
+    
+    -- 035 039: spawn 043 when defeated
+    if (mobID == 17330718 or mobID == 17330756) then
+        local southTE = GetServerVariable("[DynaXarcabard]TE43_Trigger");
+        local mobOffset = 0;
+        if (mobID == 17330756) then -- since the mobIDs aren't sequential there isn't an easy way to handle them outside of ID checks...
+            mobOffset = 1;
+        end
+        if (bit.band(southTE, bit.lshift(1, mobOffset)) == 0) then
+            southTE = bit.bor(southTE, bit.lshift(1, mobOffset));
+        end
+        if (southTE == 3) then

```

Comment:
I was about to say that since each mob can only be killed once it shouldn't matter, but then I realized this whole commit is because onMobDeath runs for every person in the group. Changes made, re-tested and it still works.